server/py/framework/utils/notifications/notification_pusher.py

```diff
@@ -226,6 +257,74 @@ def _prepare_notification_args(
         severity = alert.severity
         return message, severity
 
+    @staticmethod
+    def _prepare_notification_states(
+        notifications: list[mlrun.common.schemas.AlertNotification],
+    ) -> list[mlrun.common.schemas.NotificationState]:
+        """
+        Processes a list of alert notifications to construct a list of NotificationState objects.
+
+        Each NotificationState represents a unique type of notification (e.g., "slack", "email") and its status.
+        For each notification type, this method aggregates error messages if any notifications of that type have failed.
+        The resulting NotificationState has:
+        - An empty 'err' if all notifications of that type succeeded.
+        - An 'err' with all unique errors if all notifications of that type failed.
+        - An 'err' with unique errors if some, but not all, notifications of that type failed.
+        """
+
+        notification_errors = collections.defaultdict(
+            lambda: {
+                "errors": set(),
+                "success_count": 0,
+                "failed_count": 0,
+            },
+        )
+
+        # process each notification and gather errors by type
+        for alert_notification in notifications:
+            kind = alert_notification.notification.kind
+            reason = alert_notification.notification.reason
+
+            # count successes, failures and collect unique errors for failures
+            if reason:
+                notification_errors[kind]["errors"].add(reason)
+                notification_errors[kind]["failed_count"] += 1
+            else:
+                notification_errors[kind]["success_count"] += 1
+
+        # construct NotificationState objects based on the aggregated error data
+        notification_states = []
+        for kind, status_info in notification_errors.items():
+            errors = list(status_info["errors"])
+            success_count = status_info.get("success_count", 0)
+            failed_count = status_info.get("failed_count", 0)
+
+            if errors:
+                if success_count == 0:
+                    error_message = (
+                        f"All {kind} notifications failed. Errors: {', '.join(errors)}"
+                    )
+                else:
+                    error_message = (
+                        f"Some {kind} notifications failed. Errors: {', '.join(errors)}"
```

Comment:
is Some equals to len(errors)?. also, this could be very long, isnt it? perhaps can be truncated ? or present first error only?

Reply:
@liranbg since we use set, user will see only unique errors there. It is already truncated here https://github.com/mlrun/mlrun/blob/9ffd6dda1ee3b2d15852fb83e5addfc58e74e6d6/server/py/framework/utils/notifications/notification_pusher.py#L346

Reply:
> is Some equals to len(errors)

it is equal to summary.failed